### PR TITLE
Windows xbmgmt skeleton and cmd line parsing

### PR DESCRIPTION
**Pull request by AShivangi**
- xbmgmt version implementation
- subcommands and their options parsing
  - created stubs for xbmgmt flash options for now

### CODE REVIEW

File: src/runtime_src/core/tools/xbmgmt2/XBMgmtMain.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/**
```
Comment (stsoe) [inline src/runtime_src/core/tools/xbmgmt2/XBMgmtMain.h]: Is this file specific to xbmgmt or is it a verbatim copy from xbutil?
It latter then lets refactor to share common code.

File: src/runtime_src/core/tools/xbmgmt2/XBUtilities.cpp [hunk 2]
```diff
@@ -0,0 +1,162 @@
+/**
```
Comment (stsoe) [inline src/runtime_src/core/tools/xbmgmt2/XBUtilities.cpp]: Is this file specific to xbmgmt or is it a verbatim copy from xbutil?
It latter then lets refactor to share common code.

File: src/runtime_src/core/tools/xbmgmt2/XBUtilities.h [hunk 3]
```diff
@@ -0,0 +1,73 @@
+/**
```
Comment (stsoe) [inline src/runtime_src/core/tools/xbmgmt2/XBUtilities.h]: Is this file specific to xbmgmt or is it a verbatim copy from xbutil?
It latter then lets refactor to share common code.

File: src/runtime_src/core/tools/xbmgmt2/xbmgmt.cpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+{
+  try {
+    return main_( argc, argv );
+  } catch (const std::exception &e) {
+    xrt_core::send_exception_message(e.what(), "XBMGMT");
+  } catch (...) {
+    xrt_core::send_exception_message("Unknown error", "XBMGMT");
+  }
+  return 1;
+}
```
Comment (stsoe) [inline src/runtime_src/core/tools/xbmgmt2/xbmgmt.cpp]: Please add new line

File: src/runtime_src/core/tools/xbmgmt2/XBMgmtMain.cpp [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  } catch (po::error& e) {
+    std::cerr << "ERROR: " << e.what() << std::endl << std::endl;
+    std::cerr << globalOptions << std::endl;
+    return RC_ERROR_IN_COMMAND_LINE;
+  }
+
+  if (bOverride == false) {
+    std::cout << std::endl;
+    std::cout << "===========================================================" << std::endl;
+    std::cout << "xbutil2 is currently under development and currently is    " << std::endl;
```
Comment (stsoe) [inline src/runtime_src/core/tools/xbmgmt2/XBMgmtMain.cpp]: Looks like copied code.  Can we share?  At least "xbutil2" should be xbmgmt2

Comment (AShivangi) [reply src/runtime_src/core/tools/xbmgmt2/XBMgmtMain.cpp]: Hmm good point. Everything is the same except xbutil2->xbmgmt2. I'll refactor the code in the next PR

File: src/runtime_src/core/tools/xbmgmt2/XBMgmtMain.h (hunk 1 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/xbmgmt2/XBMgmtMain.h]: It's the exact same code. I'll refactor it in the next PR

File: src/runtime_src/core/tools/xbmgmt2/XBUtilities.cpp (hunk 2 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/xbmgmt2/XBUtilities.cpp]: I was planning to refactor this code in the next PR under a new dir tools/common

File: src/runtime_src/core/tools/xbmgmt2/XBUtilities.h (hunk 3 above)
Comment (AShivangi) [reply src/runtime_src/core/tools/xbmgmt2/XBUtilities.h]: I was planning to refactor this code in the next PR under a new dir tools/common

**Comment by AShivangi** [comment]
@stsoe I made the small changes you pointed out, but I'll do all the code refactoring in the next PR.

**Pull request merged**